### PR TITLE
Add -enable-cxx-interop flag and support for extern "C" {}

### PR DESCRIPTION
**Pull request by pschuh**
- Updated some swift headers to build with c++ and c.
- Added -enable-cxx-interop flag.
- Added support for 'extern "C"'.

### CODE REVIEW

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -518,7 +518,13 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       });
 
   } else {
-    invocationArgStrs.insert(invocationArgStrs.end(), {"-x", "c", "-std=gnu11"});
+//    invocationArgStrs.insert(invocationArgStrs.end(), {"-x", "c", "-std=gnu11"});
```
Comment (bgogul) [inline lib/ClangImporter/ClangImporter.cpp]: remove this stale comment?

File: lib/ClangImporter/SwiftLookupTable.cpp [hunk 2]
```diff
@@ -1793,21 +1797,35 @@ void importer::finalizeLookupTable(SwiftLookupTable &table,
...
+void SwiftLookupTableWriter::populateTableWithDecl(SwiftLookupTable &table,
+                                                   NameImporter &nameImporter,
+                                                   clang::Decl* decl) {
+  // Skip anything from an AST file.
+  if (decl->isFromASTFile())
+    return;
+
+  auto linkageDecl = dyn_cast<clang::LinkageSpecDecl>(decl);
+  if (linkageDecl) {
+    for (auto* decl : linkageDecl->noload_decls()) {
```
Comment (bgogul) [inline lib/ClangImporter/SwiftLookupTable.cpp]: `if (auto linkageDecl = ...)`
(If this is consistent with the style in the rest of the code.)

File: lib/ClangImporter/SwiftLookupTable.cpp [hunk 3]
```diff
@@ -1793,21 +1797,35 @@ void importer::finalizeLookupTable(SwiftLookupTable &table,
...
 }
 
+void SwiftLookupTableWriter::populateTableWithDecl(SwiftLookupTable &table,
+                                                   NameImporter &nameImporter,
+                                                   clang::Decl* decl) {
+  // Skip anything from an AST file.
+  if (decl->isFromASTFile())
+    return;
+
+  auto linkageDecl = dyn_cast<clang::LinkageSpecDecl>(decl);
```
Comment (bgogul) [inline lib/ClangImporter/SwiftLookupTable.cpp]: Add a one-line comment?

File: stdlib/public/SwiftShims/GlobalObjects.h [hunk 4]
```diff
@@ -24,7 +24,10 @@
 #include "Visibility.h"
 
 #ifdef __cplusplus
-namespace swift { extern "C" {
+#ifndef __swift__
```
Comment (bgogul) [inline stdlib/public/SwiftShims/GlobalObjects.h]: Can you document this flag somewhere?

**Review by bgogul — APPROVED** [review]
Just some minor comments...

File: stdlib/public/SwiftShims/GlobalObjects.h (hunk 4 above)
Comment (pschuh) [reply stdlib/public/SwiftShims/GlobalObjects.h]: This is the standard __swift__ version number.

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Pull request merged**